Modules/Module_Processing/Module_Population/xASL_stat_GetROIstatistics.m

```diff
@@ -488,7 +490,11 @@
 			FilePath = fullfile(x.dir.SESSIONDIR, [x.S.InputDataStrNative '.nii']);
 			if xASL_exist(FilePath, 'file')
 				Data3D = xASL_io_Nifti2Im(FilePath);
-				DataIm = xASL_im_IM2Column(Data3D,x.S.masks.WBmask);
+				if size(Data3D,4) > 1

```

Comment:
No, please don't because you make your life more difficult with the other branch that I'm working on for this. I'm saying this because your use case (3) is quite rare, and there are other places than deep down in the ROIstats thingy to check for being 4D. E.g, if you would throw an extra warning or error in the template creation, I would be fine. Or perhaps better for both of us: to put this warning or error in xASL_im_IM2Columns, with something like "4D image detected whereas the mask is not 4D, check your data, outputting NaN instead now'

Reply:
OK. I'll keep it with templates and put a warning to IM2Columns and output NaNs.